crates/bevy_text/src/font_atlas.rs

```diff
@@ -118,3 +118,139 @@ impl core::fmt::Debug for FontAtlas {
+        let containing = (1u32 << (32 - glyph_max_size.leading_zeros())).max(512);
+        font_atlases.push(FontAtlas::new(
+            textures,
+            texture_atlases,
+            UVec2::splat(containing),
+            font_smoothing,
+        ));
+
+        font_atlases.last_mut().unwrap().add_glyph(
+            textures,
+            texture_atlases,
+            physical_glyph.cache_key,
+            &glyph_texture,
+            offset,
+        )?;
+    }
+
+    Ok(get_glyph_atlas_info(font_atlases, physical_glyph.cache_key).unwrap())
+}
+
+/// Get the texture of the glyph as a rendered image, and its offset
+pub fn get_outlined_glyph_texture(
+    font_system: &mut cosmic_text::FontSystem,
+    swash_cache: &mut cosmic_text::SwashCache,
+    physical_glyph: &cosmic_text::PhysicalGlyph,
+    font_smoothing: FontSmoothing,
+) -> Result<(Image, IVec2), TextError> {
+    // NOTE: Ideally, we'd ask COSMIC Text to honor the font smoothing setting directly.
+    // However, since it currently doesn't support that, we render the glyph with antialiasing
+    // and apply a threshold to the alpha channel to simulate the effect.
+    //
+    // This has the side effect of making regular vector fonts look quite ugly when font smoothing
+    // is turned off, but for fonts that are specifically designed for pixel art, it works well.
+    //
+    // See: https://github.com/pop-os/cosmic-text/issues/279
+    let image = swash_cache
+        .get_image_uncached(font_system, physical_glyph.cache_key)
+        .ok_or(TextError::FailedToGetGlyphImage(physical_glyph.cache_key))?;
+
+    let cosmic_text::Placement {
+        left,
+        top,
+        width,
+        height,
+    } = image.placement;
+
+    let data = match image.content {
+        cosmic_text::SwashContent::Mask => {
+            if font_smoothing == FontSmoothing::None {
+                image
+                    .data
+                    .iter()
+                    // Apply a 50% threshold to the alpha channel
+                    .flat_map(|a| [255, 255, 255, if *a > 127 { 255 } else { 0 }])
+                    .collect()
+            } else {
+                image
+                    .data
+                    .iter()
+                    .flat_map(|a| [255, 255, 255, *a])
+                    .collect()
+            }
+        }
+        cosmic_text::SwashContent::Color => image.data,
+        cosmic_text::SwashContent::SubpixelMask => {
+            // TODO: implement
+            todo!()
+        }
+    };
+
+    Ok((
+        Image::new(
+            Extent3d {
+                width,
+                height,
+                depth_or_array_layers: 1,
+            },
+            TextureDimension::D2,
+            data,
+            TextureFormat::Rgba8UnormSrgb,
+            RenderAssetUsages::MAIN_WORLD,
+        ),
+        IVec2::new(left, top),
+    ))
+}
+
+/// Generates the [`GlyphAtlasInfo`] for the given subpixel-offset glyph.
+pub fn get_glyph_atlas_info(
```

Comment:
Maybe we can change the visibility to pub(crate). I don't think anything is depending on this function externally.

```suggestion
pub(crate) fn get_glyph_atlas_info(
```

Reply:
I have actually needed to use these before to write a custom text pipeline, so I'd rather leave them pub for now.

Reply:
sounds good!